operator/pkg/apis/controlplane/v1alpha1/controlplane.go

```diff
@@ -82,3 +84,8 @@ type Instances struct {
 func (c *ControlPlane) ClusterName() string {
 	return c.Name
 }
+
+type ReconcileFinalize interface {
```

Comment:
wondering if we could come up with a better name instead of `ReconcileFinalize` ? Thoughts ?

Reply:
It's more like a Lifecycle of the object, so I kept it simple to include the operations interface does. Open to suggestions.